ext/bson/bson_native.c

```diff
@@ -786,12 +788,15 @@ void pvt_put_byte( byte_buffer_t *b, const char byte)
  */
 VALUE rb_bson_byte_buffer_put_bytes(VALUE self, VALUE bytes)
 {
+  byte_buffer_t *b;
+  const char *str;
+  size_t length;
```

Comment:
Does not work.

Reply:
@saghm Const variables cannot be assigned to. Since the value of this variable depends on other statements in the function the assignment cannot happen in the declarations. Therefore the variable cannot be a const one in C.

Reply:
Makes sense! I haven't done C in a little while, so I wasn't sure if there was some compiler magic that let you assign it once if it was declared without an assignment, but I guess not. Thanks for explaining!